src/lib.rs

```diff
@@ -0,0 +1,364 @@
+//!       cause
+//!   })?;
+//!
+//! if !output.status.success() {
+//!     Err(MyStatusError {
+//!         desc: "failed to get your cargo toolchain's version",
+//!         status: output.status
+//!     })?;
+//! }
+//!
+//! println!("version was {}", String::from_utf8_lossy(&output.stdout));
+//! # Ok(())
+//! # }
+//! ```
+//!
+//!
+//! Into this:
+//!
+//! ```
+//! # use axoprocess::Cmd;
+//! # fn main() -> Result<(), Box<dyn std::error::Error>> {
+//! let output = Cmd::new("cargo", "get your cargo toolchain's version")
+//!   .arg("-V")
+//!   .output()?;
+//!
+//! println!("version was {}", String::from_utf8_lossy(&output.stdout));
+//! # Ok(())
+//! # }
+//! ```
+//!
+//! Which is, a lot nicer!
+
+// by default the "exec" commands ([`spawn`][], [`output`][], and [`status`][]):
+pub use error::*;
+use std::{
+    ffi::OsStr,
+    path::Path,
+    process::{Command, CommandArgs, CommandEnvs, ExitStatus, Stdio},
+};
+
+pub mod error;
+
+/// A fancier Command, see the crate's top-level docs!
+pub struct Cmd {
+    /// The inner Command, in case you need to access it
+    pub inner: Command,
+    summary: String,
+    log: Option<LogStrategy>,
+    check_status: bool,
+}
+
+/// Constructors
+impl Cmd {
+    /// Create a new Command with an additional "summary" of what this is trying to do
+    pub fn new(command: impl AsRef<OsStr>, summary: impl Into<String>) -> Self {
+        let inner = Command::new(command);
+        Self {
+            summary: summary.into(),
+            inner,
+            log: Some(LogStrategy::Tracing(tracing::Level::INFO)),
+            check_status: true,
+        }
+    }
+}
+
+/// Builder APIs
+impl Cmd {
+    /// Pipe stdout into stderr
+    ///
+    /// This is useful for cases where you want your program to livestream
+    /// the output of a command to give your user realtime feedback, but the command
+    /// randomly writes some things to stdout, and you don't want your own stdout tainted.
+    ///
+    /// This is equivalent to `command.stdout(std::io::stderr());` but it's easy to
+    /// not know you can do that, so this API basically exists to yell from the rooftops
+    /// that you ABSOLUTELY CAN.
+    pub fn stdout_to_stderr(&mut self) -> &mut Self {
+        self.inner.stdout(std::io::stderr());
+        self
+    }
+
+    /// Set how executions of this command should logged, accepting:
+    ///
+    /// * tracing: [`tracing::Level`][] (the default, set to `tracing::Level::INFO``)
+    /// * stdout: [`std::io::Stdout`][]
+    /// * stderr: [`std::io::Stderr`][]
+    /// * not at all: `None`
+    ///
+    /// You can explicitly invoke the selected logging with [`Cmd::log_command`][]
+    pub fn log(&mut self, strategy: impl Into<Option<LogStrategy>>) -> &mut Self {
+        self.log = strategy.into();
+        self
+    }
+
+    /// Set whether `Status::success` should be checked after executions
+    /// (except `spawn`, which doesn't yet have a Status to check).
```

Comment:
would be nice if we gave guidance on how to use spawn properly (presumably there's some API to block on the child finishing that returns a status).